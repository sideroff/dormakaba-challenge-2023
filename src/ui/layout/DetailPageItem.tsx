import Box from '@mui/material/Box';
import Typography from '@mui/material/Typography';
import { PropsWithChildren } from 'react';

interface DetailPageItemProps {
  label: string;
}

export function DetailPageItem({
  label,
  children,
}: PropsWithChildren<DetailPageItemProps>) {
  return (
    <Box>
      <Typography>
        <strong>{label}</strong>
      </Typography>
      {children}
    </Box>
  );
}
