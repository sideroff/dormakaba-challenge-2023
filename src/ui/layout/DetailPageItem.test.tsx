import { render } from '@testing-library/react';
import { DetailPageItem } from './DetailPageItem';

describe('DetailPageItem', () => {
  it('should render correctly', () => {
    const { container } = render(
      <DetailPageItem label="test">
        <div>Hello World!</div>
      </DetailPageItem>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
