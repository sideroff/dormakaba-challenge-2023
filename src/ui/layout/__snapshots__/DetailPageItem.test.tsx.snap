// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailPageItem should render correctly 1`] = `
<div
  class="MuiBox-root css-0"
>
  <p
    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
  >
    <strong>
      test
    </strong>
  </p>
  <div>
    Hello World!
  </div>
</div>
`;
